src/product_logging/spec.rs

```diff
@@ -81,6 +80,19 @@ where
     pub containers: BTreeMap<T, ContainerLogConfig>,
 }
 
+impl<T> Logging<T>
+where
+    T: Clone + Display + Ord,
+{
+    /// Get the logging configuration for `container`, falling back to the default.
+    pub fn for_container(&self, container: &T) -> Cow<ContainerLogConfig> {
```

Comment:
WDYT of calling it like this?
```suggestion
    pub fn for_container_or_default(&self, container: &T) -> Cow<ContainerLogConfig> {
```

Reply:
I'd disagree, for the consumer there should semantically be no difference between "no configuration set" or "configuration set but empty". What this `HashMap` is trying (poorly) to emulate here is something like:

```rust
struct Containers {
  #[serde(default)]
  hdfs: ContainerLogConfig,
  #[serde(default)]
  vector: ContainerLogConfig,
}
```

Reply:
Yeah, makes sense :+1: